pandas/tests/plotting/common.py

```diff
@@ -550,41 +550,59 @@ def _unpack_cycler(self, rcParams, field="color"):
         return [v[field] for v in rcParams["axes.prop_cycle"]]
 
 
-def _check_plot_works(f, filterwarnings="always", **kwargs):
+def _check_plot_works(f, filterwarnings="always", default_axes=False, **kwargs):
+    """
+    Create plot and ensure that plot return object is valid.
+    """
     import matplotlib.pyplot as plt
 
+    if default_axes:
+        gen_plots = _gen_default_plot
+    else:
+        gen_plots = _gen_two_subplots
+
     ret = None
     with warnings.catch_warnings():
         warnings.simplefilter(filterwarnings)
         try:
-            try:
-                fig = kwargs["figure"]
-            except KeyError:
-                fig = plt.gcf()
-
+            fig = kwargs.get("figure", plt.gcf())
             plt.clf()
 
-            kwargs.get("ax", fig.add_subplot(211))
-            ret = f(**kwargs)
-
-            tm.assert_is_valid_plot_return_object(ret)
-
-            if f is pd.plotting.bootstrap_plot:
-                assert "ax" not in kwargs
-            else:
-                kwargs["ax"] = fig.add_subplot(212)
-
-            ret = f(**kwargs)
-            tm.assert_is_valid_plot_return_object(ret)
+            for ret in gen_plots(f, fig, **kwargs):
+                tm.assert_is_valid_plot_return_object(ret)
 
             with tm.ensure_clean(return_filelike=True) as path:
                 plt.savefig(path)
+
+        except Exception as err:
+            raise err
         finally:
             tm.close(fig)
 
         return ret
 
 
+def _gen_default_plot(f, fig, **kwargs):
+    """
+    Create plot in a default way.
+    """
+    yield f(**kwargs)
```

Comment:
Here ``fig`` is a dummy variable.
We have it here to ensure the same function signature with ``_get_two_subplots``.

Alternatively, we can create empty figure in both generator functions and ``yield fig, f(**kwargs)``.

It seemed to me that yielding one argument rather than two is clearer, thus I did it the way it is now.